internal/cli/cli.go

```diff
@@ -198,15 +207,21 @@ func (c *RootCommand) runE(cmd *cobra.Command, args []string) error {
 	nsInformer := k8sInformerFactory.Core().V1().Namespaces()
 
 	//need at least one listener func to populate the in memory cache
-	nsInformer.Informer().AddEventHandler(k8scache.ResourceEventHandlerFuncs{
+	_, err = nsInformer.Informer().AddEventHandler(k8scache.ResourceEventHandlerFuncs{
 		AddFunc: func(new interface{}) {},
 	})
+	if err != nil {
+		klog.Log.Error(err, "error adding event handler to the namespaces informer")
+	}
 
 	coreV1Informer := k8sInformerFactory.Core().V1()
 
-	coreV1Informer.Services().Informer().AddEventHandler(k8scache.ResourceEventHandlerFuncs{
+	_, err = coreV1Informer.Services().Informer().AddEventHandler(k8scache.ResourceEventHandlerFuncs{
 		AddFunc: func(new interface{}) {},
 	})
+	if err != nil {
+		klog.Log.Error(err, "error adding event handler to the services informer")
```

Comment:
Since these now return an error; and this is in the setup; the behavior should be to `return err` and fail the startup

Reply:
added return err below